### PR TITLE
Cosmo tests

### PR DESCRIPTION
**Pull request by nstarman**
### Description

Sorts the previously unsorted tests in ``test_cosmology`` to their cosmology-class-specific test files.
The long-term goal is to refactor many of these tests to the newer test infrastructure -- many tests target a single cosmology class but more generally apply to the whole family of FLRW cosmologies.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by nstarman** [comment]
rebased to pick up CI fixes in RTD and py3.11.

**Comment by nstarman** [comment]
ping @WilliamJamieson. This is almost entirely code re-arranging, which I've done in distinct commits for clarity. I can squash when it's time to merge!

**Comment by WilliamJamieson** [comment]
`+1,765 −1,527` is a lot of changes. I'll do my best

File: astropy/cosmology/flrw/tests/test_lambdacdm.py [hunk 1]
```diff
@@ -307,6 +309,145 @@ def test_ogamma():
     assert u.allclose(cosmo.comoving_distance(z), targvals, rtol=1e-5)
 
 
+@pytest.mark.skipif(not HAS_SCIPY, reason="test requires scipy")
+def test_flat_open_closed_icosmo():
+    """Test against the tabulated values generated from icosmo.org
+    with three example cosmologies (flat, open and closed).
+    """
+
+    cosmo_flat = """\
```
Comment (WilliamJamieson) [inline astropy/cosmology/flrw/tests/test_lambdacdm.py:401]: This looks like some hard-coded data. Could this be moved to a datafile?

File: astropy/cosmology/flrw/tests/test_lambdacdm.py [hunk 2]
```diff
@@ -307,6 +309,145 @@ def test_ogamma():
...
+       3.6625000       6924.2096       1485.0852         32284.127
+       3.8250000       7045.8876       1460.2876         33996.408
+       4.0000000       7170.3664       1434.0733         35851.832
+       4.1625000       7280.3423       1410.2358         37584.767
+       4.3250000       7385.3277       1386.9160         39326.870
+       4.5000000       7493.2222       1362.4040         41212.722
+       4.6625000       7588.9589       1340.2135         42972.480
+"""
+
+    cosmo_open = """\
```
Comment (WilliamJamieson) [inline astropy/cosmology/flrw/tests/test_lambdacdm.py:436]: To data file also?

File: astropy/cosmology/flrw/tests/test_lambdacdm.py [hunk 3]
```diff
@@ -307,6 +309,145 @@ def test_ogamma():
...
+       3.6625000       6891.5474       1478.0799       32131.840
+       3.8250000       7057.4213       1462.6780       34052.058
+       4.0000000       7230.3723       1446.0745       36151.862
+       4.1625000       7385.9998       1430.7021       38130.224
+       4.3250000       7537.1112       1415.4199       40135.117
+       4.5000000       7695.0718       1399.1040       42322.895
+       4.6625000       7837.5510       1384.1150       44380.133
+"""
+
+    cosmo_closed = """\
```
Comment (WilliamJamieson) [inline astropy/cosmology/flrw/tests/test_lambdacdm.py:471]: To data file also?

**Review by WilliamJamieson — COMMENTED** [review]
Not sure I'm qualified to review these changes. The commits I did understand appear to just shift code around as described.

My only suggestion is that while doing this you might want to take the large data tables out and put them in separate data files.

File: astropy/cosmology/flrw/tests/test_lambdacdm.py (hunk 1 above)
Comment (nstarman) [reply astropy/cosmology/flrw/tests/test_lambdacdm.py:401]: Good suggestions. I'll look into this!

**Comment by nstarman** [comment]
It's a big PR, so I appreciate the review! Yes, it's almost entirely just moving code to the more appropriate test file. We had a Frankenstein file before.

File: astropy/cosmology/flrw/tests/test_lambdacdm.py (hunk 1 above)
Comment (nstarman) [reply astropy/cosmology/flrw/tests/test_lambdacdm.py]: Done!

File: astropy/cosmology/flrw/tests/test_lambdacdm.py (hunk 2 above)
Comment (nstarman) [reply astropy/cosmology/flrw/tests/test_lambdacdm.py]: ✅

File: astropy/cosmology/flrw/tests/test_lambdacdm.py (hunk 3 above)
Comment (nstarman) [reply astropy/cosmology/flrw/tests/test_lambdacdm.py]: ✅

File: astropy/cosmology/flrw/tests/test_lambdacdm.py (hunk 1 above)
Comment (nstarman) [reply astropy/cosmology/flrw/tests/test_lambdacdm.py]: I moved the data to ecsv files in ``astropy/cosmology/flrw/test/data`` and parametrized the test to cycle through the files.

Comment (nstarman) [reply astropy/cosmology/flrw/tests/test_lambdacdm.py]: I did the data file creation as a separate commit, which directly follows the commit moving the test.

Comment (nstarman) [reply astropy/cosmology/flrw/tests/test_lambdacdm.py]: I can squash it in when the PR is approved.

**Review by WilliamJamieson — APPROVED** [review]
I think these changes are fine. They appear to just refactor a base test class and then move tests around so that they are better organized.

**Comment by nstarman** [comment]
Thanks for the review!

**Comment by nstarman** [comment]
Shoot! I meant to squash these commits down.
@pllim, is there a way to quickly roll this back? Not a huge deal if not.

**Comment by pllim** [comment]
I'd rather not touch the history in `main`. I don't see any merge commits here, so probably ok to leave as is.

**Comment by nstarman** [comment]
Ok. Thanks!

**Pull request merged**